### ISSUE TITLE
EN_setlinknodes produces an error when a PRV/PSV valve is reversed

### ISSUE THREAD

**Issue by LRossman** [OpenWaterAnalytics/EPANET#561]
This is related to the comment I left on Issue #40 but I'm creating a new issue here for the sake of bug tracking. If one tries to reverse the orientation of a PRV or PSV between two junction nodes (neither of which is connected to another valve) using the `EN_setlinknodes` function, it returns with error 220 - illegal valve connection to another valve. The code snippet below will produce the error:
```
    EN_addnode(ph, "J1", EN_JUNCTION, &j1);
    EN_addnode(ph, "J2", EN_JUNCTION, &j2);
    EN_addlink(ph, "PRV1", EN_PRV, "J1", "J2", &prv1);
    err = EN_setlinknodes(ph, prv1, j2, j1);
    printf("\n  PRV1 reversed - err = %d\n", err);
```
The problem is caused by the `valvecheck `function in `project.c` when it tries to check for a valid connection between a valve and itself. The fix is to remove this type of check.

**Comment by samhatchett** [comment]
I think maybe the fix should be to test for this node-reversal edge case. Removing the check might have bad side effects, since most of the time I figure you _do_ actually want to enforce the check in some way?

**Comment by LRossman** [comment]
There are two types of checks made in `valvecheck `for a given valve. First is a check to see if it's connected to a Tank/Reservoir. That's not at issue here. Then a check is made to see if the valve is illegally connected to another valve. It's when looping through each of the "other" valves that the valve in question gets checked against itself which causes the erroneous error. Like I said, the fix is to modify the loop's code so that it doesn't try to check connections to itself. E.g. if we have valves V1, V2, and V3 and we call `EN_setlinknodes` for V1, then V1 should only be checked against V2 and V3 and not V1 against V1 (which is what happens now).

**Comment by samhatchett** [comment]
I like the solution of passing in the link index (or zero) as an extra parameter.

**Comment by LRossman** [comment]
This issue has been resolved via PR #562 .
